classes/task/RemoveFailedJobs.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/**
+ * @file classes/task/RemoveFailedJobs.php
+ *
+ * Copyright (c) 2022 Simon Fraser University
+ * Copyright (c) 2022 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class RemoveFailedJobs
+ * @ingroup tasks
+ *
+ * @brief Remove the much older failed jobs form the failed list
+ */
+
+namespace PKP\task;
+
+use APP\facades\Repo;
+use Carbon\Carbon;
+use PKP\config\Config;
+use PKP\scheduledTask\ScheduledTask;
+
+class RemoveFailedJobs extends ScheduledTask
+{
+    /**
+     * @copydoc ScheduledTask::getName()
+     */
+    public function getName()
+    {
+        return __('admin.scheduledTask.removeFailedJobs');
+    }
+
+
+    /**
+     * @copydoc ScheduledTask::executeActions()
+     */
+    public function executeActions()
+    {
+        $cleanUpPeriod = (int) Config::getVar('queues', 'failed_job_clean_period', null);
```

Comment:
The last `null` (the default value) should probably be the same as the default you're introducing into the `config.TEMPLATE.inc.php` file. That way folks to install a fresh application will get the same default as those who upgrade but forget to add the new `config.inc.php` option.

Reply:
Done .